crates/store/src/state.rs

```diff
@@ -96,17 +96,114 @@ pub struct TransactionInputs {
     pub found_unauthenticated_notes: BTreeSet<NoteId>,
 }
 
+#[derive(Debug)]
+pub struct Blockchain(Mmr);
+
+impl Blockchain {
+    pub fn get(&self) -> &Mmr {
+        &self.0
+    }
+    pub fn get_mut(&mut self) -> &mut Mmr {
+        &mut self.0
+    }
+
+    pub fn get_latest_block_number(&self) -> BlockNumber {
+        let block_number: u32 = (self.0.forest() - 1)
+            .try_into()
+            .expect("chain_mmr always has, at least, the genesis block");
+
+        block_number.into()
+    }
+
+    pub fn get_chain_length(&self) -> usize {
+        let latest = self.get_latest_block_number();
+        latest.as_usize() + 1
+    }
```

Comment:
Nit: In Rust the `get_` prefix is usually omittted, so `latest_block_number` and `chain_length`.